src/main/java/apoc/load/LoadLdap.java

```diff
@@ -0,0 +1,256 @@
+package apoc.load;
+
+import apoc.ApocConfiguration;
+import com.novell.ldap.*;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Mode;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import java.io.UnsupportedEncodingException;
+import java.util.*;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class LoadLdap {
+
+    @Procedure(name = "apoc.load.ldap", mode = Mode.READ)
+    @Description("apoc.load.ldap(\"key\" or {connectionMap},{searchMap}) Load entries from an ldap source (yield entry)")
+    public Stream<LDAPResult> ldapQuery(@Name("connection") final Object conn, @Name("search") final Map<String,Object> search) {
+
+        LDAPManager mgr = new LDAPManager(getConnectionMap(conn));
+
+        return mgr.executeSearch(search);
+    }
+
+    public static Map<String, Object> getConnectionMap(Object conn) {
+        if (conn instanceof String) {
+            //String value = "ldap.forumsys.com cn=read-only-admin,dc=example,dc=com password";
+            Object value = ApocConfiguration.get("loadldap").get(conn.toString() + ".config");
+            // format <ldaphost:port> <logindn> <loginpw>
+            if (value == null) throw new RuntimeException("No apoc.loadldap."+conn+".config ldap access configuration specified");
+            Map<String, Object> config = new HashMap<>();
+            String[] sConf = ((String) value).split(" ");
+            config.put("ldapHost", sConf[0]);
+            config.put("loginDN", sConf[1]);
+            config.put("loginPW", sConf[2]);
+
+            return config;
+
+        } else {
+            return (Map<String,Object> ) conn;
```

Comment:
ldap:// url is not convenient it has also a kind of search syntax in it (so it is more than a connection alone) I only need the connection parameters, search we do via the 'searchMap'